questionpy_server/cache.py

```diff
@@ -111,34 +127,35 @@ def put(self, key: str, value: bytes) -> Path:
             raise SizeError(f"Item itself exceeds maximum allowed size of {self.max_bytes} bytes",
                             max_size=self.max_bytes, actual_size=size)
 
-        # Update `_total_bytes` depending on whether the key existed already or not.
-        if key in self._files:
-            self._total_bytes -= self._files[key].size
-        self._total_bytes += size
+        async with self._lock:
+            # Save the bytes on filesystem.
+            path = self.directory / (key + self._extension)
+            tmp_path = path.parent / (path.name + self._tmp_extension)
 
-        # Save the bytes on filesystem.
-        path = self.directory / (key + self._extension)
+            if size != await to_thread(tmp_path.write_bytes, value):
+                raise IOError("Failed to write bytes to file")
```

Comment:
Versuche mal in der `test_put` Funktion mittels [unittest.mock](https://docs.python.org/3/library/unittest.mock.html#module-unittest.mock) `Path.write_bytes` zu mocken, damit ein anderer Wert als `size` zurückgegeben wird und auf diese Exception geprüft wird.